.changeset/chilly-dryers-begin.md

```diff
@@ -0,0 +1,47 @@
+---
+"@opennextjs/cloudflare": patch
+---
+
+introduce new `initOpenNextCloudflareForDev` utility and make `getCloudflareContext` synchronous
+
+introduce a new `initOpenNextCloudflareForDev` function that when called in the [Next.js config file](https://nextjs.org/docs/app/api-reference/config/next-config-js) integrates the Next.js dev server with the open-next Cloudflare adapter. Most noticeably this enables `getCloudflareContext` to work in
+the edge runtime (including middlewares) during development.
+
+Moving forward we'll recommend that all applications include the use of this utility in their config file (there is no downside in doing so and it only effect local development).
+
+Example:
+
+```js
+// next.config.mjs
+
+import { initOpenNextCloudflareForDev } from "@opennextjs/cloudflare";
+
+/** @type {import('next').NextConfig} */
+const nextConfig = {};
+
+initOpenNextCloudflareForDev();
+
+export default nextConfig;
+```
+
+thanks to this new utility we are also able to make `getCloudflareContext` synchronous, so `await`ing such function is no longer necessary
+
+Example:
+
+- Before:
+
+  ```js
+  import { getCloudflareContext } from "@opennextjs/cloudflare";
+
+  const cloudflareContext = await getCloudflareContext();
```

Comment:
We still should not call this at top level as we are not in a request context

Reply:
it's just an example, I didn't want to get too verbose...

would `// ...` lines solve the issue?
or would you prefer a more verbose example?
or maybe I can remove the example altogether since this is also pretty obvious anyways

Reply:
I don't like the fact that the example is broken.

As the changeset is already long, I would rather shorten it and remove the example but add details to our documentation.

Reply:
example removed 👍